cmd/nerdctl/namespace.go

```diff
@@ -114,7 +125,130 @@ func namespaceLsAction(cmd *cobra.Command, args []string) error {
 			numVolumes = len(volEnts)
 		}
 
-		fmt.Fprintf(w, "%s\t%d\t%d\t%d\n", ns, numContainers, numImages, numVolumes)
+		labels, err := client.NamespaceService().Labels(ctx, ns)
+		if err != nil {
+			return err
+		}
+		for k, v := range labels {
+			labelStrings = append(labelStrings, strings.Join([]string{k, v}, "="))
+		}
+		sort.Strings(labelStrings)
+		fmt.Fprintf(w, "%s\t%d\t%d\t%d\t%v\t\n", ns, numContainers, numImages, numVolumes, strings.Join(labelStrings, ","))
 	}
 	return w.Flush()
 }
+
+func newNamespaceRmCommand() *cobra.Command {
+	namespaceRmCommand := &cobra.Command{
+		Use:           "remove [OPTIONS] NAMESPACE [NAMESPACE...]",
+		Aliases:       []string{"rm"},
+		Short:         "Remove one or more namespaces",
+		RunE:          namespaceRmAction,
+		SilenceUsage:  true,
+		SilenceErrors: true,
+	}
+	namespaceRmCommand.Flags().BoolP("cgroup", "c", false, "delete the namespace's cgroup")
+	return namespaceRmCommand
+}
+
+func namespaceRmAction(cmd *cobra.Command, args []string) error {
+	if len(args) == 0 {
+		return errors.New("please specify a namespace")
+	}
+	var exitErr error
+	client, ctx, cancel, err := newClient(cmd)
+	if err != nil {
+		return err
+	}
+	defer cancel()
+	opts, err := deleteOpts(cmd)
+	if err != nil {
+		return err
+	}
+	namespaces := client.NamespaceService()
+
+	for _, target := range args {
+		if err := namespaces.Delete(ctx, target, opts...); err != nil {
+			if !errdefs.IsNotFound(err) {
+				if exitErr == nil {
+					exitErr = errors.Wrapf(err, "unable to delete %v", target)
+				}
+				log.G(ctx).WithError(err).Errorf("unable to delete %v", target)
+				continue
+			}
+		}
+		fmt.Println(target)
+	}
+	return exitErr
+}
+
+func newNamespaceCreateCommand() *cobra.Command {
+	namespaceCreateCommand := &cobra.Command{
+		Use:           "create NAMESPACE [<key>=<value]",
+		Aliases:       []string{"c"},
+		Short:         "Create a new namespace",
+		RunE:          namespaceCreateAction,
+		SilenceUsage:  true,
+		SilenceErrors: true,
+	}
+	namespaceCreateCommand.Flags().BoolP("cgroup", "c", false, "delete the namespace's cgroup")
+	return namespaceCreateCommand
+}
+
+func namespaceCreateAction(cmd *cobra.Command, args []string) error {
+	if len(args) == 0 {
+		return errors.New("please specify a namespace")
+	}
+	client, ctx, cancel, err := newClient(cmd)
+	if err != nil {
+		return err
+	}
+	defer cancel()
+	if err != nil {
+		return err
+	}
+	namespace, labels := ObjectWithLabelArgs(args)
```

Comment:
labels should be taken from `--label`

Reply:
now usage is  "nerdctl ns label namespace1 key1=value1",like docker tag command,
i think it is not necessary to use cmd.Flags().GetString("label") to get lable value

Reply:
```
nerdctl namespace create foo --label key1=val1 --label key2=val2

nerdctl namespace update foo --label key3=val3 --label key4=val4
```